src/common/reactor.cc

```diff
@@ -360,23 +346,37 @@ namespace Pistache::Aio
 
             void removeAll()
             {
+                GUARD_AND_DBG_LOG(handlers_arr_mutex_);
+                
                 index_ = 0;
                 handlers.fill(NULL);
             }
 
             std::shared_ptr<Handler> operator[](size_t index) const
             {
-                return handlers.at(index);
+                return handlers.at(index); // calls const "at"
             }
 
-            std::shared_ptr<Handler> at(size_t index) const
-            {
+            std::shared_ptr<Handler> at(size_t index)
+            { // It's const, except that the mutex is locked and unlocked
                 if (index >= index_)
                     throw std::runtime_error("Attempting to retrieve invalid handler");
-
+                GUARD_AND_DBG_LOG(handlers_arr_mutex_);
+                
                 return handlers.at(index);
             }
 
+            std::shared_ptr<Handler> at(size_t index) const
+            {
+                // We cast away the "const" of "this" before calling the
+                //  non-const version of "at". The non-const "at" is not
+                //  declared const because it locks and unlocks a mutex; but it
+                //  is const in that the class instance is in the same state
+                //  after the call vs. before (mutex is unlocked again before
+                //  non-const "at" exits) - so this is OK.
+                return(((HandlerList *)this)->at(index));
```

Comment:
Please avoid using C-style casts. In this case I think `const_cast` would be more appropriate. Also, you could consider @kiplingw's suggestion; making a mutex mutable is a common pattern.

Reply:
Good eye @Tachi107. I missed that.